[Release PR] V0.1.1

Miniplop opened the pull request:
<!--- Provide a general summary of your changes in the title above -->

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- If it resolves an open issue, please link to the issue here. For example "Resolves: #137" -->

## Checklist:

<!--- Put an `x` in all the boxes that apply. -->
<!--- If your change requires a documentation PR, please link it appropriately -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

<!--- A cute animal picture is welcome to close your PR! -->

Miniplop committed:
remove the important for overlay modal background color + clean some modal components

Miniplop committed:
Create a Slicemachine modal to override default react modal style instead of using !important properties in CSS

Miniplop committed:
design the review modal

MarcMcIntosh committed:
add test to core plus extract Auth from core

MarcMcIntosh committed:
WIP: prompt: existing repo or create a new one

Heyrows committed:
big refacto of Core file structure, add login step of init, add submodules on Core

Heyrows committed:
add missing confirmation

Miniplop committed:
Make the modal appear only once + chnage the design

MarcMcIntosh committed:
WIP: ls-repos merge with sm-init-script

Miniplop committed:
add tracking route on server side + call this route on submit

MarcMcIntosh committed:
test(sm-core): add some tests for reponame validation

Miniplop committed:
display the comment only if the user select 5 less rating

MarcMcIntosh committed:
add tests to init

MarcMcIntosh committed:
test(sm-init): set up jest and typescript

MarcMcIntosh committed:
sm-init: refactor checking for repos

Miniplop committed:
display the review modal

Miniplop committed:
fix some lint issues

Miniplop committed:
Convert IconButton to tsx

MarcMcIntosh committed:
chore: fix compiling issues

Miniplop committed:
Convert the Container to tsx + fix tests

MarcMcIntosh committed:
replace node-ffetch with axios

MarcMcIntosh committed:
fix: replace node-fetch wit axios

MarcMcIntosh committed:
fix(core): add node_modules back into jest transformIgnore

MarcMcIntosh committed:
chore(sm-init): downgrade ora

Miniplop committed:
fix some other linter errors

MarcMcIntosh committed:
fix: server exiting early

MarcMcIntosh committed:
fix: response from validate session endpoint is a string

Miniplop committed:
Fix more linter errors

MarcMcIntosh committed:
fix: order of repos and prompt order

MarcMcIntosh committed:
remomve console.log

MarcMcIntosh committed:
fix: remove unimplemted code

MarcMcIntosh committed:
remove console.log statement

MarcMcIntosh committed:
temp fix:(sm-init): remove listeners before prompting

Miniplop committed:
remove old navbar + old unused push-all route

Miniplop committed:
migrate server/index file + parse-oembed file to typescript

Miniplop committed:
remove unused lib push

Miniplop committed:
Clean some lint errors

Heyrows committed:
add command flags

Miniplop committed:
Update jest to version 27.0 + fix audit command to be properly scoped to dev dep and ignore the security issue on copy-template-dir dep

Miniplop committed:
use yarn improved audit for all the package to avoid getting error code when a issue is not high level

Miniplop committed:
use yarn improved audit for all the package to avoid getting error code when a issue is not high level

Heyrows committed:
after review: modify authConfig parameter name, remove event listener on login confirmation

MarcMcIntosh committed:
test(sm-init): add tests for maybe create repo stage

MarcMcIntosh committed:
test(sm-init): repo name display

MarcMcIntosh committed:
chore(sm-init): remove comment

MarcMcIntosh committed:
chore(sm-init): fix linter warnings

MarcMcIntosh committed:
chore(sm-init): update readme

MarcMcIntosh committed:
chore(sm-init): resync with sm-init-script branch

Heyrows committed:
fix linter core & init

MarcMcIntosh committed:
resync with sm-init-script

MarcMcIntosh committed:
chore(sm-core): fix some linter issues

MarcMcIntosh committed:
chore(resync): re-sync again

MarcMcIntosh committed:
chore(lint): remove createCore import

MarcMcIntosh committed:
fix: hapi warning about return types

Heyrows committed:
add test for login, signup and handler data, don't transpile tests file anymore

Heyrows committed:
Update jest to version 27.0 + fix audit command to be properly scoped to dev dep and ignore the security issue on copy-template-dir dep

Heyrows committed:
remove commentary on jest configs

MarcMcIntosh committed:
chore(core): add function to create repo

Miniplop committed:
Clean some components

Miniplop committed:
create the start / status and post auth routes for login process

Miniplop committed:
create the start / status and post auth routes for login process

Heyrows committed:
bypass login if valid session, use ora to display check

MarcMcIntosh committed:
WIP: create repository from sm-init

MarcMcIntosh committed:
fix: add user-agent when creating a repo

MarcMcIntosh committed:
chore: linter warnings

MarcMcIntosh committed:
chore: run prettier

MarcMcIntosh committed:
ci: add cache back in

MarcMcIntosh committed:
fix: remove console.log

MarcMcIntosh committed:
chore(sm-init): remove ora

MarcMcIntosh committed:
chore(sm-init): remove chalk

MarcMcIntosh committed:
chore(resync): resync with sm-init-script

Heyrows committed:
bypass login if valid session, use ora to display check

MarcMcIntosh committed:
test(core) add a couple more tests to the communication module

MarcMcIntosh committed:
chore(sm-core): re exprt communication module

MarcMcIntosh committed:
resync with ls-repos

MarcMcIntosh committed:
fix(core test): remember to import jest

MarcMcIntosh committed:
chore(sm-init): fix test

Heyrows committed:
fix jest types, add tests for validating session helper

MarcMcIntosh committed:
chore: add additional test for creating repo

MarcMcIntosh committed:
chore: run prettier

Miniplop committed:
Make the poll works

Heyrows committed:
review: change test names and remove console.log

MarcMcIntosh committed:
feat(sm-init): add frame wrok detection

Miniplop committed:
display the login modal only on 402 push custom types

MarcMcIntosh committed:
fix: linter warnings

Miniplop committed:
add polling on the login and signup process

MarcMcIntosh committed:
test(core-framwork): check all known framewroks

MarcMcIntosh committed:
fix(core files): on't create a varable in try/catch block

MarcMcIntosh committed:
fix(sm init): frame work detection messages

Miniplop committed:
remove add sm script

Miniplop committed:
Merge pull request #190 from prismicio/init-bypass-login

[V0.1.1] - AAuser, I want SM init to recognise if i'm logged in, so that I can bypass login

Miniplop committed:
Merge pull request #186 from prismicio/sm-init-script

[V0.1.1] Implement slicemachine-core & sm-init packages.

MarcMcIntosh committed:
chore(prettier): run prettier

MarcMcIntosh committed:
chore(ls-repos): resync with sm-init-script

Miniplop committed:
add cors authorization for the localhost api + design the modal

MarcMcIntosh committed:
chore(init and ccore): remove rules from eslint

MarcMcIntosh committed:
chore(sm-core): refactor poll

MarcMcIntosh committed:
chore(sm-init): make repos list more pretty

Miniplop committed:
display error/success message on login

MarcMcIntosh committed:
chore(resync): resync with ls-repos and lint

MarcMcIntosh committed:
chore(rsync): resync with sm-init-create-repo

Miniplop committed:
display the modal on the push slice if the user is not connected

Miniplop committed:
Use the local package instead of distant core package

Miniplop committed:
add build test for sm-init and the core

MarcMcIntosh committed:
fix(list-repos): sorting and formting

MarcMcIntosh committed:
chore(ls-repos): refactoring

MarcMcIntosh committed:
chore(ls-repos): refactoring

MarcMcIntosh committed:
chore(ls-repos): refactor

MarcMcIntosh committed:
chore(resync): resync with ls-repos branch

MarcMcIntosh committed:
refactor(ls-repos): defaulting to zero for the list option works.

MarcMcIntosh committed:
refactor: rename promptForRepo

MarcMcIntosh committed:
refactor(ls-repos): use template strings

MarcMcIntosh committed:
fix(ls-repos): typo

MarcMcIntosh committed:
refactor(ls-repos): add return type to validateName

MarcMcIntosh committed:
refactor(ls-repos): minor changes

MarcMcIntosh committed:
refacctor(ls-repos): test enums

MarcMcIntosh committed:
fix(ls-repos): fix tests

MarcMcIntosh committed:
refactor(ls-repos): add io-ts

MarcMcIntosh committed:
chore(ls-repos sm-core): lint tests

MarcMcIntosh committed:
chore(ls-repos sm-init): add tests to linter

MarcMcIntosh committed:
chore(sm-init): throw errors

MarcMcIntosh committed:
refactor(core) cocmmunication repo reader

MarcMcIntosh committed:
refactor(core): add function maybeRepoNameFromSMFile

MarcMcIntosh committed:
refactor(sm-init): extract canUpdateCustomTypes

MarcMcIntosh committed:
refactor(core utils roles): move roles in to utils

MarcMcIntosh committed:
fix some issues with io-ts and rpl

MarcMcIntosh committed:
fix(ls-repos): display the hostname only

MarcMcIntosh committed:
fix(roles): fix the test for new roles

MarcMcIntosh committed:
refactor(ls-repos): refore base from core listrepos

MarcMcIntosh committed:
fix(ls-repos): add double quotes to repo address

MarcMcIntosh committed:
resync sm-init-create-repo with ls-repos

Miniplop committed:
use npm for sm ui

Miniplop committed:
use npm for sm ui

Miniplop committed:
use npm in the CI

Miniplop committed:
use npm for the core

Miniplop committed:
use npm for the core

Miniplop committed:
use npm for sm-init

Miniplop committed:
use npm for sm-init

Miniplop committed:
use npm for sm-commons

Miniplop committed:
remove improved yarn audit from the sm-init package

Miniplop committed:
remove improved yarn audit from the sm-init package

Miniplop committed:
use npm for vue-slicezone

Miniplop committed:
use npm for nuxt-sm

Miniplop committed:
use npm for next-slicezone

Miniplop committed:
Use npm with lerna

Miniplop committed:
Create a makefile to ease the clean / install process

MarcMcIntosh committed:
Merge branch 'sm-init-create-repo' into sm-init-detect-framework

Miniplop committed:
Use npm for the test project

Miniplop committed:
Add command to run the linter in the precommit hook

Miniplop committed:
remove package.lock and package.json from prettier scope

Miniplop committed:
use the prod url for prismic tracking service

Miniplop committed:
Merge pull request #187 from prismicio/kje/add-review-modal

AAU, when I have pushed at least 1 slice and 1 custom type, I see a modal asking for my review on the SM

Miniplop committed:
add io-ts and fp-ts in the package.json

Miniplop committed:
add lerna install

Miniplop committed:
use npx in the sm install

MarcMcIntosh committed:
refactor(detect framework): remove extra catcches and dead code

Miniplop committed:
build he core in the install command + return empty json response on auth routes

MarcMcIntosh committed:
refactor detect frame work to use less catches

MarcMcIntosh committed:
refactor detect frame diffrent messages on fail

MarcMcIntosh committed:
fix(sm-init): remove if condition on install sm

MarcMcIntosh committed:
chore(yarn.lock): updated for some reason

Miniplop committed:
Design + wording

MarcMcIntosh committed:
chore(exisint-repo): add hint

Miniplop committed:
change the start script to redirect to the sm init if needed

MarcMcIntosh committed:
chore(ls-repos): add hint to crete repo

MarcMcIntosh committed:
resync with ls-repos

MarcMcIntosh committed:
Merge branch 'sm-init-create-repo' into sm-init-detect-framework

Miniplop committed:
add audit fix script on every project

Miniplop committed:
review changes

Miniplop committed:
review changes

MarcMcIntosh committed:
sm-init-create-repo: make required changes

Miniplop committed:
migrate cookie parsing into the core

MarcMcIntosh committed:
refactor(core communication crete repo): error handling

Miniplop committed:
change the setAuthConfig

Miniplop committed:
merge the error code management for fake client and auth error

MarcMcIntosh committed:
resync with sm-init-create-repo

Miniplop committed:
review changes

Miniplop committed:
Update package-lock.json to work on node 12

Miniplop committed:
the lib validate npm packages is not supported since 4 years so we cannot update the dep

MarcMcIntosh committed:
fix(list repos): parameter order

MarcMcIntosh committed:
Merge branch 'sm-init-create-repo' into sm-init-detect-framework

MarcMcIntosh committed:
fix(detect-repo): handle edge case 'gatsby'

MarcMcIntosh committed:
Merge pull request #188 from prismicio/ls-repos

Ls repos

MarcMcIntosh committed:
fix(detect-frame-work): remove comments

MarcMcIntosh committed:
refactor(detectFramework): add isUnsupported function

MarcMcIntosh committed:
core(framework): add return types

Heyrows committed:
init: reorganise steps and add comments

Miniplop committed:
review changes

Miniplop committed:
add a link above the signin link to go manualy to the dashbaord

MarcMcIntosh committed:
fix(create-repo): handle upper case letters

MarcMcIntosh committed:
chore(create-repo): add test case for repos names with a capital letter

MarcMcIntosh committed:
Merge pull request #189 from prismicio/sm-init-create-repo

Sm init create repo

MarcMcIntosh committed:
chore(resync): resync detect-reppo with v0.1.1

Miniplop committed:
Update the lock files with new dep

Miniplop committed:
use npm cache instead of yarn cache for the CI

Miniplop committed:
Add node version check before everything else

Heyrows committed:
add configure project step, add first test mocking the part needed in the core

MarcMcIntosh committed:
Merge pull request #192 from prismicio/sm-init-detect-framework

feat(sm-init): add frame wrok detection

Miniplop committed:
Merge branch 'v0.1.1' into kje/login-sm-ui

Miniplop committed:
Merge pull request #193 from prismicio/kje/login-sm-ui

AAUser, when my connexion token expires, I can login from SM UI

Heyrows committed:
add tests for configure project, add new end message of the init script

Miniplop committed:
Update tests with new FrameworkEnum

Miniplop committed:
Change the login modal retry link wording

Miniplop committed:
Use node 1 for building the app / installing packages

Miniplop committed:
Merge pull request #194 from prismicio/kje/start-script

AAUser, I want to launch SM on local host 9999 when I run SM start

Heyrows committed:
add test for install sm step

Heyrows committed:
Merge branch 'v0.1.1' into ac/init-configure-project

Heyrows committed:
fix detect framework tests

Heyrows committed:
remove useless sm.json

Heyrows committed:
Merge pull request #199 from prismicio/ac/init-configure-project

AAuser, I want SM to be already configured after downloading SM in terminal

Heyrows committed:
fix linter

Heyrows committed:
fixing test

Heyrows committed:
not crash on malformed apiEndpoint, keep native error to have more info

Heyrows committed:
Merge branch 'v0.1.1' into ac/init-configure-project

Heyrows committed:
add test to cover fully the get repo name from manifest

Heyrows committed:
Merge pull request #202 from prismicio/ac/init-configure-project

[V0.1.1] Init - configure project

Miniplop committed:
Add source slice-machine into login dashboard url

Miniplop committed:
WIP release : change the code package name and the init package name

Heyrows committed:
synchronize init, ui and core versions

Miniplop committed:
rename slicemachine core in the ui dep

Heyrows committed:
add user agent to slicemachine ui server

Heyrows committed:
Merge branch 'release' of github.com:prismicio/slice-machine into release

Heyrows committed:
add missing lock file

Miniplop committed:
deactivate the review modal

Miniplop committed:
Merge branch 'release' of github.com:prismicio/slice-machine into release

Miniplop committed:
update init readme

Miniplop committed:
Add core readme

Heyrows committed:
fix cookies

Miniplop committed:
Add preview image

Miniplop committed:
Merge pull request #204 from prismicio/release

Changes for the V0.1.1 release

MarcMcIntosh committed:
chore(ct-api): conditionally add user-agent when running server side

MarcMcIntosh committed:
chore(linter DefaultClient): fix linter warnings

MarcMcIntosh committed:
fix(default-client): fix some of the type checking

Miniplop committed:
Add a package description

Miniplop committed:
Add libraries path when building the sm.json

Heyrows committed:
beta versions

Heyrows committed:
Merge branch 'v0.1.1' of github.com:prismicio/slice-machine into v0.1.1

MarcMcIntosh committed:
refactor(slice-machine-ui): remove commented out catch

Miniplop committed:
Merge pull request #208 from prismicio/kje/add-default-librairies

Kje/add default librairies

Miniplop committed:
create a changelog for the V0.1.1

MarcMcIntosh committed:
remove user-agent condition

hypervillain commented on packages/slice-machine/server/src/api/custom-types/index.ts:
```diff
@@ -1,21 +1,25 @@
   return matches.reduce((acc: Array<CustomType<ObjectTabs>>, p: string) => {
     const key = path.basename(path.dirname(p));
-    const pathTopreview = path.join(path.dirname(p), "index.png");
+    const pathToPreview = path.join(path.dirname(p), "index.png");
```
should we allow more extensions?

Miniplop committed:
Update the changelog

MarcMcIntosh committed:
fix(user-agent): remoev old implimentation

MarcMcIntosh committed:
Merge pull request #207 from prismicio/mm/user-agent

AADev, when A user send request to the CT API, I send "slice-machine" as "User-Agent" header in the request

MarcMcIntosh committed:
fix(core version): slice-machine-ui uses the wrong core version

Miniplop committed:
restore version to 0.1.1

Miniplop committed:
restore to version 0.1.0

Miniplop committed:
Publish

 - next-slicezone@0.1.3-alpha.0
 - nuxt-sm@0.0.9-alpha.0
 - slice-machine-ui@0.1.1-alpha.0
 - @slicemachine/core@0.1.1-alpha.0
 - sm-commons@0.0.24-alpha.0
 - @slicemachine/init@0.1.1-alpha.0
 - vue-slicezone@0.0.27-alpha.0

Miniplop committed:
Clean changelog

Miniplop committed:
Merge pull request #209 from prismicio/kje/add-v0.1.1-changelog

create a changelog for the V0.1.1
